### ISSUE TITLE
Example demonstrating multi-agent system using smolagents

### ISSUE THREAD

**Issue by haesleinhuepf** [huggingface/smolagents#13]
Hi again @aymeric-roucher ,

I just wrote [an example notebook](https://scads.github.io/generative-ai-notebooks/66_arxiv_agent/multiagent_write_review.html) about how to let multiple smolagents write a manuscript together and I'm curious about two things:
- Would this example notebook be in scope for your repository as example? I could clean it a bit up (depending in your feedback) and send a PR if this makes sense.
- Is there room for improvement? I'm in particular curious if the implementation of the scheduler agent makes sense.

Feedback welcome!

Best,
Robert

**Comment by rishabhmehrotra** [comment]
@haesleinhuepf great notebook, thanks for sharing -- it was quite helpful to see multiple agents interacting via the scheduler. the `prompt_scadsai_llm` function in the notebook makes the setup less self-contained to run independently. I'm guessing `prompt` [calls to the function](https://github.com/ScaDS/generative-ai-notebooks/blob/main/docs/66_arxiv_agent/arxiv_utilities.py#L243C5-L243C23) in the Scientific writer and reviewer agents can easily be replaced by an LLM call reusing smolagents components? This might make the notebook self-contained.

**Comment by haesleinhuepf** [comment]
> I'm guessing `prompt` [calls to the function](https://github.com/ScaDS/generative-ai-notebooks/blob/main/docs/66_arxiv_agent/arxiv_utilities.py#L243C5-L243C23) in the `Scientific writer` and `reviewer` agents can easily be replaced by an LLM call reusing smolagents components? This might make the notebook self-contained.

Good point! In order to do this, it would be good if #54 would get merged.



> it was quite helpful to see multiple agents interacting via the scheduler.

I noticed that my custom scheduler implementation is similar to (and not as well developed) as the [Managed Agents Example](https://github.com/huggingface/smolagents/blob/main/docs/source/en/examples/multiagents.md). I will try to update the notebook to use this infrastructure.

**Comment by aymeric-roucher** [comment]
@haesleinhuepf thank you for proposing this notebook! It's a good idea to have such a workflow.
However I think the notebook could be simplified:
- distribute_sub_task, agent_factory, create_agent: IMO these functions can be removed for simplification
